### DISCUSSION TITLE
Framework/Engine Nominations

### DISCUSSION THREAD

**Discussion by dimensionscape** [haxecollab/ludum-dare-56#1] [Ideas]
Please nominate an engine or framework. This is the place to make your case why **XXXX** should be used. Based on the nominations here we will(maybe?) have as many as 3 rounds of votes starting September 14th to determine the most popular choice. 

Making a firm decision ahead of time will give participants time to setup, familiarize and create necessarily boiler-plate with the technology chosen.

Ideally candidates for this should be well documented or have helpful resources and examples available for the public. Although not exclusively necessary, the engine or framework should probably support a web target, which generally makes the entire process less burdensome for the build process, deployment, play-testing, and safe access for players/judges. 

(Note: If there is a clear winner in the first vote, we will not need to consider runoff contingencies or multiple voting rounds)

### Currently Nominated

1. HaxeFlixel
2. Ceramic
3. Raylib-hx

## Update(9/15)
Everyone who is participating in the collab should submit their contributor PR ASAP for their vote to be counted.

**Comment by EyeDaleHim** [comment]
I'll start off with the most popular choice - [HaxeFlixel](https://haxeflixel.com/)

HaxeFlixel is powered on Lime and OpenFL. Which means we also leverage use of these two frameworks.

I say HaxeFlixel is the most popular choice because there are more projects created in HaxeFlixel than OpenFL these days, so I figure it be best to leverage the amount of people using HaxeFlixel.

It also suits this criteria - well-made documentation, helpful resources, examples, and can support the web target.

**Comment by l0go** [comment]
[Ceramic](https://ceramic-engine.com/)

It is a modern framework that is pretty simple to pick up. It has a pretty opinionated API, that feels a lot more "complete" than Flixel. Additional the creator, Jeremyfa, has expressed an interest in contributing music to this entry.

The documentation is pretty solid too, at least for Haxe standards.

For the record, Flixel is probably a better choice as most are familiar with it. I just think Ceramic should be in the running

**Comment by CCobaltDev** [comment]
Flixel

**Comment by LeonGamerPS1** [comment]
Flixel

**Comment by mikaib** [comment]
Raylib.

Pros:
- Very flexible, can allow things considered "quite complex" to be done pretty easily.
- More control, i also see this as a con as it means you need to setup up more of a foundation for the game.
- Very fun to use
- Could be very informative for some newer game devs
- Generally pretty easy to pick up

Cons:
- More control (see above why)
- No "states" or "sprites" which most of the community is familliar with, but you can make those yourself pretty easily.

**Reply by EyeDaleHim** [reply]
Perhaps you're forgetting to provide a link?

**Reply by Just-Feeshy** [reply]
I assume by RayLib, you're mentioning:
https://github.com/foreignsasquatch/raylib-hx

**Reply by mikaib** [reply]
That is the most likely option yes.

**Comment by jeremyfa** [comment]
I’m obviously biased and would suggest Ceramic, but other options are good too, as long as they are not too difficult to pick up for newcomers in the related tech. Having the setup done for each participant before the jam seems crucial too

**Reply by Just-Feeshy** [reply]
Wait a minute.. lol
